### PR TITLE
*: add comments to all exported members

### PR DESCRIPTION
**Pull request by erizocosmico**
Also made `golint`, `go vet` and `gofmt -s` happy and simplified some code.

### CODE REVIEW

**Comment by erizocosmico** [comment]
Done, can be reviewed now

File: sql/expression/aggregation.go [hunk 1]
```diff
@@ -34,15 +40,18 @@ func (c *Count) Resolved() bool {
 	return c.Child.Resolved()
 }
 
+// Name returns the of the node.
```
Comment (mcarmonaa) [inline sql/expression/aggregation.go]: s/returns the of/returns the name of

**Comment by erizocosmico** [comment]
Depends on #19 so tests pass

File: sql/expression/aggregation.go (hunk 1 above)
Comment (erizocosmico) [reply sql/expression/aggregation.go]: good catch

**Comment by erizocosmico** [comment]
Fixed @mcarmonaa

**Pull request merged**